Blocks/Persistence/file.h

```diff
@@ -92,35 +94,49 @@ class IteratorOverFileOfPersistedEntries {
 // The implementation of a persister based exclusively on appending to and reading one text flie.
 template <typename ENTRY>
 class FilePersister {
+ protected:
+  // { last_published_index + 1, last_published_us + 1us }, or { 0, 0us } for an empty persister.
+  struct end_t {
+    uint64_t index;
+    std::chrono::microseconds us;
+  };
+  static_assert(sizeof(std::chrono::microseconds) == 8, "");
+  static_assert(sizeof(end_t) == 16, "");
+
  private:
   struct Impl {
+    // Just `std::atomic<end_t> end;` won't work in g++ until 5.1, ref.
+    // http://stackoverflow.com/questions/29824570/segfault-in-stdatomic-load/29824840#29824840
+    // std::atomic<end_t> end;
+    // Actually, the above line would work fine on my machine -- D.K. -- but only if `std::atomic<end_t> end`
+    // is the first member declaration in the class. I'll leave it commented out for now.
+    current::atomic_that_works<end_t> end;
+
     const std::string filename;
-    idxts_t next_initializer;
-    std::atomic<uint64_t> next_index;
-    std::atomic<std::chrono::microseconds> next_timestamp;
     std::ofstream appender;
 
     Impl() = delete;
     explicit Impl(const std::string& filename)
-        : filename(filename),
-          next_initializer(ValidateFileAndInitializeNext(filename)),
+        : end(ValidateFileAndInitializeNext(filename)),
+          filename(filename),
           appender(filename, std::ofstream::app) {
-      next_index = next_initializer.index;
-      next_timestamp = next_initializer.us;
       assert(appender.good());
     }
 
-    // Replay the file but ignore its contents. Used to initialize `next_{index,timestamp}` at startup.
-    static idxts_t ValidateFileAndInitializeNext(const std::string& filename) {
+    // Replay the file but ignore its contents. Used to initialize `end` at startup.
+    static end_t ValidateFileAndInitializeNext(const std::string& filename) {
       std::ifstream fi(filename);
       if (fi.good()) {
         IteratorOverFileOfPersistedEntries<ENTRY> cit(fi);
         while (cit.ProcessNextEntry([](const idxts_t&, const char*) {})) {
-          ;  // Read through all the lines, let `IteratorOverFileOfPersistedEntries` maintain `next`.
+          // Read through all the lines.
+          // Let `IteratorOverFileOfPersistedEntries` maintain its own `next_`, which later becomes `this->end`.
+          ;
         }
-        return cit.Next();
+        const auto& end = cit.Next();
+        return end_t{end.index, end.us};
       } else {
-        return idxts_t();
+        return end_t{0ull, std::chrono::microseconds(0)};
```

Comment:
Rule of zero: http://en.cppreference.com/w/cpp/language/rule_of_three

I figured for `atomic`-s it's more important than a proper constructor.